src/zarr/core/common.py

```diff
@@ -162,3 +164,10 @@ def parse_order(data: Any) -> Literal["C", "F"]:
     if data in ("C", "F"):
         return cast(Literal["C", "F"], data)
     raise ValueError(f"Expected one of ('C', 'F'), got {data} instead.")
+
+
+def parse_dtype(dtype: Any) -> np.dtype[Any]:
```

Comment:
This type could be greatly improved. Really, we want the same signature as `np.dtype`, but with an overload for `str -> dtype(object)`. But maybe as a follow up.